.golangci.yml

```diff
@@ -1,79 +1,124 @@
+        list-mode: lax
+        deny:
+          - pkg: "flag"
+            desc: '`flag` package is only allowed in main.go'
+          - pkg: "io/ioutil"
+            desc: '`io/ioutil` package is deprecated, use the `io` and `os` package instead'
+          # TODO: Prevent using these without a reason
+          # - pkg: "reflect"
+          #   desc: '`reflect` package is dangerous to use'
+          # - pkg: "unsafe"
+          #   desc: '`unsafe` package is dangerous to use'
+
   errcheck:
-     check-type-assertions: true
-     check-blank: true
-     disable-default-exclusions: true
-     exclude-functions:
-       - '(*bytes.Buffer).Write' # always returns nil error
-       - '(*github.com/valyala/bytebufferpool.ByteBuffer).Write' # always returns nil error
-       - '(*github.com/valyala/bytebufferpool.ByteBuffer).WriteByte' # always returns nil error
-       - '(*github.com/valyala/bytebufferpool.ByteBuffer).WriteString' # always returns nil error
+    check-type-assertions: true
+    check-blank: true
+    disable-default-exclusions: true
+    exclude-functions:
+      - '(*bytes.Buffer).Write' # always returns nil error
+      - '(*github.com/valyala/bytebufferpool.ByteBuffer).Write' # always returns nil error
+      - '(*github.com/valyala/bytebufferpool.ByteBuffer).WriteByte' # always returns nil error
+      - '(*github.com/valyala/bytebufferpool.ByteBuffer).WriteString' # always returns nil error
 
   errchkjson:
     report-no-exported: true
 
   exhaustive:
+    check-generated: true
     default-signifies-exhaustive: true
 
   forbidigo:
     forbid:
-      - ^(fmt\.Print(|f|ln)|print|println)$
-      - 'http\.Default(Client|Transport)'
+      - ^print(ln)?$
+      - ^fmt\.Print(f|ln)?$
+      - ^http\.Default(Client|ServeMux|Transport)$
       # TODO: Eventually enable these patterns
-      # - 'time\.Sleep'
-      # - 'panic'
+      # - ^panic$
+      # - ^time\.Sleep$
+    analyze-types: true
+
+  gci:
+    sections:
+      - standard
+      - prefix(github.com/gofiber/fiber)
+      - default
+      - blank
+      - dot
+      # - alias
+    custom-order: true
+
+  goconst:
+    numbers: true
 
   gocritic:
+    # TODO: Uncomment the following lines
+    # enabled-tags:
+    #   - diagnostic
+    #   - style
+    #   - performance
+    #   - experimental
+    #   - opinionated
     disabled-checks:
-      - ifElseChain
+      - ifElseChain # TODO: Do not disable
+      - hugeParam
+      - rangeExprCopy
+      - rangeValCopy
+    settings:
+      captLocal:
+        paramsOnly: false
+      elseif:
+        skipBalanced: false
+      underef:
+        skipRecvDeref: false
+      # NOTE: Set this option to false if other projects rely on this project's code
+      unnamedResult:
+        checkExported: false
 
   gofumpt:
     module-path: github.com/gofiber/fiber
     extra-rules: true
 
   gosec:
     excludes:
-      - G104 # Provided by errcheck
+      - G104 # TODO: Enable this again. Mostly provided by errcheck
```

Comment:
Do you know which cases are not handled by errcheck?

Reply:
I don't know off-head right now and don't have access to a Go system at the moment. Simply try it yourself and drop the line :D

Reply:
In the current codebase it doesn't catch anything that other linters don't catch, but I'd like to see some cases that it does catch which the others dont.
Actually, I just remembered another reason why I disabled it - golangci-lint has a bug that prevents you from setting global ignores for this rule https://github.com/golangci/golangci-lint/issues/3749
So until that get's fixed, you'd have to manually add ignores everywhere, which adds a lot of noise to the code.